src/__tests__/element-queries.js

```diff
@@ -801,4 +801,70 @@ test('get/query textarea element by current value', () => {
   expect(queryByDisplayValue('World').id).toEqual('content-textarea')
 })
 
+test('DescriptionOf uses aria-describedby', () => {
+  const {
+    getByRole,
+    getDescriptionOf,
+    queryDescriptionOf,
+    findDescriptionOf,
+  } = renderIntoDocument(`
+    <main>
+      <button aria-describedby="button-description">Click me</button>
+      <p id="button-description">Hello, Dave!</p>
+    </main>
+  `)
+
+  expect(getDescriptionOf(getByRole('button'))).toHaveTextContent(
+    'Hello, Dave!',
+  )
+  expect(queryDescriptionOf(getByRole('button'))).toHaveTextContent(
+    'Hello, Dave!',
+  )
+  expect(findDescriptionOf(getByRole('button'))).resolves.toHaveTextContent(
```

Comment:
Make this test async, and await this assertion. Otherwise it resolves/rejects after the test completes.

Reply:
This would do the same it's doing now, no? Otherwise wouldn't jest fail because it doesn't resolve?

Reply:
At least it's how I understood https://jestjs.io/docs/en/asynchronous.html#resolves-rejects

Reply:
Try this:

```js
expect(Promise.reject()).resolves.toBe(true)
```

You'll see that the test passes, but then there's a promise rejection in the output.

Reply:
Yeah, I prefer to not return. Too easy to forget to do if you change the rest later. Just use async/await.

Reply:
>  Too easy to forget to do if you change the rest later

Already forgot it without anything else...so yeah :smile:

Reply:
Done